### PR TITLE
Create testbench to check viash version switching

### PR DESCRIPTION
**Pull request by Grifs**
Move code from main function to new 'outerMain' that contains normal logic code and leave try/catch & system.exit code in regular main
Add error handling for bad file URL in IO object. 404s would cause the child process to hang and throw an uncatchable exception

### CODE REVIEW

File: src/main/scala/io/viash/helpers/IO.scala [hunk 1]
```diff
@@ -126,7 +128,18 @@ object IO {
...
+      // Prevent hanging when resource is not available
+      val connection = url.openConnection().asInstanceOf[HttpURLConnection]
+      connection.setConnectTimeout(5000)
+      connection.setReadTimeout(5000)
+      connection.connect()
+
+      if (connection.getResponseCode >= 400)
+        throw new RuntimeException(s"Could not download file: ${uri.toString} , Response code: ${connection.getResponseCode}")
+      else
+        (url #> path.toFile).!!
```
Comment (rcannood) [inline src/main/scala/io/viash/helpers/IO.scala:141]: Can we use the connection to download the file? Or is this happening implicitly already? (Seems unlikely to me)

**Review by rcannood — CHANGES_REQUESTED** [review]
Two minor remarks, and merge #330 first before merging this PR.

Otherwise LGTM

File: src/test/scala/io/viash/auxiliary/MainRunVersionSwitch.scala [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      }
+    }
+    
+    val stdout = outStream.toString
+    val stderr = errStream.toString
+
+    // any output to the console is handled in a separate process where we don't have access to anymore
+    // so we expect empty outputs
+    assert(stdout.isEmpty())
+    assert(stderr.isEmpty())
```
[the remark this answers is not in the document]
Comment (Grifs) [reply src/test/scala/io/viash/auxiliary/MainRunVersionSwitch.scala]: Good idea, it works nicely.
`
Process(...).!(ProcessLogger(Console.out.println, Console.err.println))
`

File: src/main/scala/io/viash/helpers/IO.scala (hunk 1 above)
Comment (Grifs) [reply src/main/scala/io/viash/helpers/IO.scala]: new version:

**Pull request merged**